Sources/NIO/SystemDarwin.swift

```diff
@@ -0,0 +1,443 @@
+            if err != EINTR {
+                preconditionIsNotBlacklistedErrno(err: err, where: #function)
+                throw IOError(errnoCode: err, reason: "close")
+            }
+        }
+    }
+
+    @inline(never)
+    public static func bind(descriptor: CInt, ptr: UnsafePointer<sockaddr>, bytes: Int) throws {
+         _ = try syscall(blocking: false) {
+            Darwin.bind(descriptor, ptr, socklen_t(bytes))
+        }
+    }
+
+    @inline(never)
+    @discardableResult
+    // TODO: Allow varargs
+    public static func fcntl(descriptor: CInt, command: CInt, value: CInt) throws -> CInt {
+        return try syscall(blocking: false) {
+            Darwin.fcntl(descriptor, command, value)
+        }.result
+    }
+
+    @inline(never)
+    public static func socket(domain: CInt, type: CInt, `protocol`: CInt) throws -> CInt {
+        return try syscall(blocking: false) {
+            return Darwin.socket(domain, type, `protocol`)
+        }.result
+    }
+
+    @inline(never)
+    public static func setsockopt(socket: CInt, level: CInt, optionName: CInt,
+                                  optionValue: UnsafeRawPointer, optionLen: socklen_t) throws {
+        _ = try syscall(blocking: false) {
+            Darwin.setsockopt(socket, level, optionName, optionValue, optionLen)
+        }
+    }
+
+    @inline(never)
+    public static func getsockopt(socket: CInt, level: CInt, optionName: CInt,
+                                  optionValue: UnsafeMutableRawPointer,
+                                  optionLen: UnsafeMutablePointer<socklen_t>) throws {
+        _ = try syscall(blocking: false) {
+            Darwin.getsockopt(socket, level, optionName, optionValue, optionLen)
+        }.result
+    }
+
+    @inline(never)
+    public static func listen(descriptor: CInt, backlog: CInt) throws {
+        _ = try syscall(blocking: false) {
+            Darwin.listen(descriptor, backlog)
+        }
+    }
+
+    @inline(never)
+    public static func accept(descriptor: CInt,
+                              addr: UnsafeMutablePointer<sockaddr>?,
+                              len: UnsafeMutablePointer<socklen_t>?) throws -> CInt? {
+        let result: IOResult<CInt> = try syscall(blocking: true) {
+            let fd = Darwin.accept(descriptor, addr, len)
+
+            #if !os(Linux)
+                if fd != -1 {
+                    do {
+                        try Posix.fcntl(descriptor: fd, command: F_SETNOSIGPIPE, value: 1)
+                    } catch {
+                        _ = Darwin.close(fd) // don't care about failure here
+                        throw error
+                    }
+                }
+            #endif
+            return fd
+        }
+
+        if case .processed(let fd) = result {
+            return fd
+        } else {
+            return nil
+        }
+    }
+
+    @inline(never)
+    public static func connect(descriptor: CInt, addr: UnsafePointer<sockaddr>, size: socklen_t) throws -> Bool {
```

Comment:
why are all these now duplicated? Couldn't we do a `SystemPosix.swift` instead of having so much duplication between `SystemLinux/Darwin`?

Reply:
These aren't really POSIX per se (some of them are BSD - the sockets one), and some of them are C standard (ISO), and some are POSIX (ISO).  If we want split them up across those boundaries, thats okay, but I think that this is *far* more legible and much easier to see what is going on than all the indirection.

This became even more important as I started adding a third path through this code.  It was extremely difficult to reason about the changes and follow along as I was making the changes to support MSVCRT.

Reply:
but the code is identical for Linux & Mac, right? Regarding the naming: I'm happy with `SystemUNIX` or whatever but I'd like to avoid duplicating the whole file for Linux/Mac. For the _actual_ differences I think a `SystemDarwin`/`SystemLinux` does make sense but stuff like `connect` is the same for both, right?

Reply:
I think that the aliasing through sys... makes it far less readable.  I suppose we could just `#if os()` the implementations.

Reply:
Okay, so, I've opted instead to split out `System` into `System*` and `BSDSocket`.  There is duplication in `BSDSocket` still, but, collapsing those cases can be done and we can judge which is more readable.

This basically comes at a cost: you get a much larger change in a single commit.  But, the refactoring will just take a single larger step rather than the multiple intermediate ones.